catalyst/contrib/dl/callbacks/trace.py

```diff
@@ -0,0 +1,138 @@
+from typing import Union  # isort:skip
+from operator import lt, gt
+
+import warnings
+
+from pathlib import Path
+
+from catalyst.dl import Callback, CallbackOrder, State
+from catalyst.dl.utils import (
+    trace_model_from_state,
+    save_traced_model,
+)
+
+
+class TracerCallback(Callback):
+
+    def __init__(
+            self,
+            metric_key: str,
+            stage: str = None,
+            mode: str = "max",
+            method_name: str = "forward",
+            requires_grad: bool = False,
+            opt_level: str = None,
+            loader: Union[str, int] = None,
+            trace_mode: str = "eval",
+            out_dir: Union[str, Path] = None,
+            out_model: Union[str, Path] = None,
+    ):
+        """
+        Traces model using created experiment and runner.
+        
+        Args:
+            metric_key (str): Metric key we should trace model based on
+            mode (str): Metric max or min value affects tracing.
+            method_name (str): Model's method name that will be
+                used as entrypoint during tracing
+            stage (str): Stage from experiment from which model and loader
+                will be taken
+            loader (str): Loader name to get the batch from
```

Comment:
why do we need `stage` and `loader` specification? 
this callback can be used with specified `stage` already and you can use `state.input` for batch input during tracing